### PR TITLE
Fixed QUnit on Android (closes #112)

### LINKED ISSUE #112: Fix QUnit tests in Android browser
https://saucelabs.com/tests/fb5739b4a4484cc3a3aa8deffdd1a946

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

**Comment by inikulin** [comment]
Please, review without me

File: test/client/fixtures/runner/automation/regression-test.js [hunk 1]
```diff
@@ -262,10 +262,9 @@ $(document).ready(function () {
                 .addClass(TEST_ELEMENT_CLASS)
                 .click(function () {
                     clicked = true;
-                })
-                .appendTo('body');
+                });
 
-        window.setTimeout(function () {
+        $iframe[0].onload = function () {
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/automation/regression-test.js]: Please use the [waitForIframe](https://github.com/AlexanderMoskovkin/qunit-harness#wait-for-an-iframe-action) function. It's availabe in `qunit-harness` yet.

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 2]
```diff
@@ -1361,10 +1429,27 @@ $(document).ready(function () {
         select.selectedIndex = 0;
         equal(select.selectedIndex, 0);
 
-        clickPlaybackAutomation($option[0], {}, function () {
-            equal(select.selectedIndex, 2);
-            equal(handlersLog.join(), browserUtils.isIE ? 'select mousedown,select mouseup,select change,select click' : 'option mousedown,option mouseup,select change,option click');
-            startNext();
+        window.async.series({
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/automation/select-element-test.js]: As I see the test for `select with size more than one`. Should we ignore this for iOS and Android

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 3]
```diff
@@ -1375,14 +1460,40 @@ $(document).ready(function () {
         //bind handlers
         bindSelectAndOptionHandlers($(select), $option);
 
-        clickPlaybackAutomation($option[0], {}, function () {
-            equal(select.selectedIndex, 2);
+        window.async.series({
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/automation/select-element-test.js]: the same as above

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 4]
```diff
@@ -412,6 +412,43 @@ $(document).ready(function () {
         }, 1500);
     });
 
+    // NOTE: Android and iOS ignores size and multiple attributes, all select elements behave like select with size=1
+    if (browserUtils.isSafari && browserUtils.hasTouchEvents || browserUtils.isAndroid) {
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/automation/select-element-test.js]: let's move it to the var, something like `ignoreMultisizeSelectTests`. And we can write the comment when the var is declared once

File: src/client/ui/select-element.js [hunk 5]
```diff
@@ -67,7 +67,9 @@ function createOption (option, $parent) {
 
             if (browserUtils.isFirefox || browserUtils.isIE)
                 eventSimulator.mousedown(browserUtils.isFirefox ? option : curSelectEl);
-            eventSimulator.mouseup(browserUtils.isFirefox ? option : curSelectEl);
+
+            if(!browserUtils.isAndroid)
```
Comment (AlexanderMoskovkin) [inline src/client/ui/select-element.js]: @AndreyBelym Could you explain me why we need this condition?

Comment (AndreyBelym) [inline src/client/ui/select-element.js]: These events are not raised on Android (and iOS too) in native browser.

**Comment by AndreyBelym** [comment]
FPR

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 6]
```diff
@@ -1395,7 +1505,7 @@ $(document).ready(function () {
         select.selectedIndex = 0;
 
         clickPlaybackAutomation(select, {}, function () {
-            equal(handlersLog.join(), browserUtils.isIE ? 'select mousedown,select mouseup,select click' : 'select mousedown,select mouseup,select click');
+            equal(handlersLog.join(), 'select mousedown,select mouseup,select click');
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/automation/select-element-test.js]: the previous variant was quite funny =)

**Comment by AlexanderMoskovkin** [comment]
lgtm

\cc @helen-dikareva please review

**Comment by helen-dikareva** [comment]
lgtm 
@VasilyStrelyaev please check new test names

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 7]
```diff
@@ -412,6 +414,43 @@ $(document).ready(function () {
         }, 1500);
     });
 
+    // NOTE: Android and iOS ignores size and multiple attributes, all select elements behave like select with size=1
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
    // NOTE: Android and iOS ignore the size and multiple attributes, all select elements behave like select with size=1
```

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 8]
```diff
@@ -412,6 +414,43 @@ $(document).ready(function () {
...
+    if (isMobileBrowser) {
+
+        asyncTest('click on option  raises error in select with size more than one when option list is collapsed', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
        asyncTest('in select elements with 'size' more than 1, click on an option raises an error when the option list is collapsed', function () {
```

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 9]
```diff
@@ -412,6 +414,43 @@ $(document).ready(function () {
...
+            setTimeout(function () {
+                equal(currentErrorType, ERROR_TYPE.invisibleActionElement);
+                equal(currentErrorElement, '&lt;option class=&quot;testElement&quot;&gt;');
+                equal(currentActionSourceIndex, 312);
+
+                startNext();
+            }, 1500);
+        });
+
+        asyncTest('click on option  raises error in select with multiple attribute when option list is collapsed', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
        asyncTest('in select elements with the 'multiple' attribute, click on an option raises an error when option list is collapsed', function () {
```

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 10]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
     });
 
+    // NOTE: Android and iOS ignore size and multiple attributes, all select elements behave like select with size=1
+    if (isMobileBrowser) {
 
-    module('mouse actions with select element with attribute size more than one');
-    asyncTest('click on option', function () {
-        var select = createSelect(2);
+        module('mouse actions with multiline select element');
+        asyncTest('click on select element with size attribute greater than one and on option', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
        asyncTest('click on the 'select' element with the 'size' attribute greater than one, then click on an option', function () {
```

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 11]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
+            clickPlaybackAutomation(select, {}, function () {
+                ok(selectElement.isOptionListExpanded($(select)));
+                startNext();
+            });
         });
-    });
 
-    asyncTest('click on option with scroll', function () {
-        var select = createSelect(2);
+        asyncTest('click on select element with multiple attribute and on option', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
        asyncTest('click on the 'select' element with the 'multiple' attribute, then click on an option', function () {
```

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 12]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
+            select.selectedIndex = 0;
+            equal(select.selectedIndex, 0);
+            clickPlaybackAutomation($option[0], {}, function () {
+                equal(select.selectedIndex, 0);
+                startNext();
+            });
         });
-    });
 
+        asyncTest('click on option (select size more than options count)', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
        asyncTest('click on an option (the select size is more than the option count)', function () {
```

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 13]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
-        select.selectedIndex = 0;
-        $(select).focus();
+        module('press actions with select element with attribute size more than one');
+        asyncTest('press down/up in select', function () {
+            var select = createSelect();
 
-        pressDownUpKeysActions(select);
-    });
+            $(select).attr('size', '2');
+            //NOTE: IE11 set selectedIndex = -1 after setting attribute size != 1
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: //NOTE: IE11 sets selectedIndex = -1 after setting attribute size != 1

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 14]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
-        //NOTE: IE11 set selectedIndex = -1 after setting attribute size != 1
-        select.selectedIndex = 0;
-        $(select).focus();
+        asyncTest('press right/left in select', function () {
+            var select = createSelect();
 
-        pressRightLeftKeysActions(select);
-    });
+            $(select).attr('size', '2');
+            //NOTE: IE11 set selectedIndex = -1 after setting attribute size != 1
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: ...sets..

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 15]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
-        //NOTE: IE11 set selectedIndex = -1 after setting attribute size != 1
-        select.selectedIndex = 0;
-        $(select).focus();
+        asyncTest('press down/up in select with "size" more than options count', function () {
+            var select = createSelect();
 
-        pressDownUpKeysActions(select);
-    });
+            $(select).attr('size', '10');
+            //NOTE: IE11 set selectedIndex = -1 after setting attribute size != 1
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: ...sets..

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 16]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
-    asyncTest('press down/up in select with "size" more than options count', function () {
-        var select = createSelect();
+            pressRightLeftKeysActions(select);
+        });
 
-        $(select).attr('size', '10');
-        //NOTE: IE11 set selectedIndex = -1 after setting attribute size != 1
-        select.selectedIndex = 0;
-        $(select).focus();
+        asyncTest('press down/up in select with "size" more than options count', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: asyncTest('press down/up in select with "size" more than the option count', function () {

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 17]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
-    asyncTest('press right/left in select with "size" more than options count', function () {
-        var select = createSelect();
+            pressDownUpKeysActions(select);
+        });
 
-        $(select).attr('size', '10');
-        //NOTE: IE11 set selectedIndex = -1 after setting attribute size != 1
-        select.selectedIndex = 0;
-        $(select).focus();
+        asyncTest('press right/left in select with "size" more than options count', function () {
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
        asyncTest('press right/left in select with "size" more than the option count', function () {
```

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 18]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
+            var select = createSelect();
 
-        pressRightLeftKeysActions(select);
-    });
+            $(select).attr('size', '10');
+            //NOTE: IE11 set selectedIndex = -1 after setting attribute size != 1
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: ...sets...

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 19]
```diff
@@ -591,139 +630,164 @@ $(document).ready(function () {
...
-    asyncTest('click on option with scroll', function () {
-        var select = createSelect();
 
-        select.selectedIndex = 0;
-        equal(select.selectedIndex, 0);
-        $(select).attr('multiple', 'multiple');
-        clickPlaybackAutomation($(select).children(':last')[0], {}, function () {
-            equal(select.selectedIndex, 4);
-            startNext();
+        module('mouse actions with select element with attribute multiple');
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: module('mouse actions with the "select" element with the "multiple" attribute');

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 20]
```diff
@@ -960,215 +1024,217 @@ $(document).ready(function () {
...
-        select.selectedIndex = 0;
-        $(select).focus();
-
-        pressRightLeftKeysActionsForSelectWithOptgroups(select, startNext, browserUtils.isWebKit);
-    });
-
+   // NOTE: Android ignores size and multiple attributes, all select elements behave like select with size=1
+   if (!isMobileBrowser) {
+
+       module('mouse actions with select with option groups and size more than one');
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
       module('mouse actions with select with the "groups" option and size more than one');
```

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 21]
```diff
@@ -960,215 +1024,217 @@ $(document).ready(function () {
...
+       asyncTest('click optgroup', function () {
+           var select  = createSelectWithGroups(),
+               $select = $(select),
+               group   = $select.find('optgroup')[0];
+
+           $select.attr('size', '4');
+           select.selectedIndex = 0;
+
+           window.setTimeout(function () {
+               //NOTE: during set selected option IE and Mozilla scroll select
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: //NOTE: when setting the selected option, IE and Mozilla scroll the select

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 22]
```diff
@@ -960,215 +1024,217 @@ $(document).ready(function () {
...
+           $select.css({
+               position: 'absolute',
+               top:      '200px',
+               left:     '300px'
+           });
+           $select.attr('size', '5');
+           select.selectedIndex = 0;
+
+           window.setTimeout(function () {
+               //NOTE: during set selected option IE and Mozilla scroll select
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: //NOTE: when setting the selected option, IE and Mozilla scroll the select

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 23]
```diff
@@ -960,215 +1024,217 @@ $(document).ready(function () {
...
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: //NOTE: when setting the selected option, IE and Mozilla scroll the select

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 24]
```diff
@@ -960,215 +1024,217 @@ $(document).ready(function () {
...
+               $optgroup  = $select.find('optgroup').eq(2),
+               $newOption = $('<option></option>').text('outer option')
+                   .addClass(TEST_ELEMENT_CLASS)
+                   .insertAfter($optgroup);
+
+           $select.attr('size', '5');
+           select.selectedIndex = 0;
+
+           window.setTimeout(function () {
+               //NOTE: during set selected option IE and Mozilla scroll select
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: //NOTE: when setting the selected option, IE and Mozilla scroll the select

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 25]
```diff
@@ -960,215 +1024,217 @@ $(document).ready(function () {
...
+           $select.css({
+               position: 'absolute',
+               top:      '200px',
+               left:     '300px'
+           });
+           $select.attr('size', '5');
+           select.selectedIndex = 0;
+
+           window.setTimeout(function () {
+               //NOTE: during set selected option IE and Mozilla scroll select
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]: //NOTE: when setting the selected option, IE and Mozilla scroll the select

File: test/client/fixtures/runner/automation/select-element-test.js [hunk 26]
```diff
@@ -960,215 +1024,217 @@ $(document).ready(function () {
...
+                   equal(select.selectedIndex, 9);
+                   window.setTimeout(function () {
+                       startNext();
+                   }, 0);
+               });
+           }, 100);
+       });
+
+
+       module('press actions with select with option groups and size more than one');
```
Comment (VasilyStrelyaev) [inline test/client/fixtures/runner/automation/select-element-test.js]:
```
       module('press actions with select with the "groups" option and size more than one');
```

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by AndreyBelym** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
The grammar remarks are not fixed
/r-

File: test/client/fixtures/runner/automation/regression-test.js [hunk 27]
```diff
@@ -262,10 +262,9 @@ $(document).ready(function () {
                 .addClass(TEST_ELEMENT_CLASS)
                 .click(function () {
                     clicked = true;
-                })
-                .appendTo('body');
+                });
 
-        window.setTimeout(function () {
+        $iframe[0].addEventListener('load', function () {
```
Comment (AlexanderMoskovkin) [inline test/client/fixtures/runner/automation/regression-test.js]: Sorry for this, but we've fixed the `waifForIframe` function in `qunit-harness`. Please try to use it again

**Comment by AndreyBelym** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by VasilyStrelyaev** [comment]
lgtm

**Pull request merged**